[T7][W09-B4]

dylansyj opened the pull request:
Created a UserGuide V0.0 for doTASK project

limmlingg committed:
Edit product name

dylansyj committed:
Created a V0.0 User Guide for doTASK

dylansyj committed:
Updated User Guide V0.0

dylansyj committed:
Added features into UserGuideV0.0

dylansyj committed:
Added FAQ into UserGuideV0.0

dylansyj committed:
testing

dylansyj committed:
test

dylansyj committed:
Merge pull request #1 from CS2103JAN2017-W09-B4/Testing_Branch

Testing branch

dylansyj committed:
Fix indentation issues in UserGuideV0.0

dylansyj committed:
asd

dylansyj committed:
Fix indentation (should be alright now)

dylansyj committed:
Fix Indentation for Features, UserGuideV0.0

dylansyj committed:
Fixing indentations are fun

dylansyj committed:
Indentation fix

dylansyj committed:
Indentation fix please let it be the last

dylansyj committed:
Updated Dev Guide v0.0 : User Stories

dylansyj committed:
Edited DevGuide

dylansyj committed:
Categorized user stories in DevGuide

dylansyj committed:
Fixed indentations for both UserGuideV0.0 and DevGuide.

dylansyj committed:
Attached the UI image to the UserGuideV0.0.

dylansyj committed:
Add Priority_screen.png into UserGuideV0.0

dylansyj committed:
Edit image file in UserGuideV0.0, wrong name

limmlingg committed:
Update User Guide with screenshots

limmlingg committed:
Change User Guide name

limmlingg committed:
Add newline to User Guide

dylansyj committed:
Edit Features in UserGuide, 2.1 - 2.6.

dylansyj committed:
Fix indentation

dylansyj committed:
Fix indentation

dylansyj committed:
Fix Indentation

dylansyj committed:
Fix indentation
Edit Features

limmlingg committed:
Update screenshots

limmlingg committed:
Update screenshots and image names

eevyern committed:
Added documentation for commands delete, viewpriority, complete

qixiang-xu committed:
Added Track overall progress feature, updated FAQ

Some minor additions such as examples to certain features.

dylansyj committed:
Edit some overlapped features.
Fix indentation.
Updated command summary

dylansyj committed:
Fix Indentation
Minor edits

dylansyj committed:
Add search feature into the user guide.
All requirements of V0.0 met.

dylansyj committed:
Fix indentation for userguide

dylansyj committed:
Fix Indentation - Travis

limmlingg committed:
Update AboutUs and README

dylansyj committed:
Add Undo, Redo feature. Missed out in V0.0

dylansyj committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-W09-B4/main

# Conflicts:
#	docs/UserGuide.md

dylansyj committed:
Refactor Parser (incomplete)
Comment on Model ( parts that needs refactoring to suit logic )

To all : Please refer to user guide to standardize commands

dylansyj committed:
Minor refactor in model and logic component

dylansyj committed:
Done with add feature for Logic
Halfway into edit feature

dylansyj committed:
Done refactoring for Edit Featuer - Logic component

dylansyj committed:
Search Feature implemented in Logic

limmlingg committed:
Update user guide URL for help window

qixiang-xu committed:
Refactoring Model

Deadline, Information, PriorityLevel, ReadOnlyTask, Task, TaskName &
UniqueTagList.

dylansyj committed:
Renamed one of the duplicate UniqueTagList to UniqueTaskList.
Fix issues in logic/model

hwkchia commented on docs/UserGuide (original).md:
```diff
@@ -0,0 +1,186 @@
+# AddressBook Level 4 - User Guide
```
If for any reason you want to keep the original guide for reference, may I suggest that you don't stage it?

dylansyj replied:
Ok, we will remove it. Thank you

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+> <img src="images/add_screen.jpg" width="600">
+
+
+5. After step 6, you can view your task in 3 different tabs.
```
If you look at the rendered version, this IS step 6.

dylansyj replied:
sorry for the mistake, we will amend it. Thank you!

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+
+   > **Today/Date**
+   >
+   > * Shows you the list of tasks you have for the specified date. By default, it will show the tasks for today. Click left/right arrow to navigate between the dates.
```
What do you mean by "click left/right arrow"?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+   > **Today/Date**
+   >
+   > * Shows you the list of tasks you have for the specified date. By default, it will show the tasks for today. Click left/right arrow to navigate between the dates.
+   > <img src="images/today_screen.jpg" width="600">
```
And where's the newly added CS3230 assignment on the mockup?  It also does not appear in the "Priority" and "Overview" tabs?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+   >
+   > **Overview**
+   >
+   > * Shows you an overall view of the dates. The highest priority tasks will be shown as a preview for every single date. Scroll to navigate between the dates.
```
Scroll?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+   >
+   > **Priority**
+   >
+   > * Shows you 4 options of priority tabs, from highest to lowest, which are indicated by different colours. Selecting the tab allows you to view the list of tasks that is under that priority. Eg. By clicking on â€œHighest Priorityâ€� tab, it will bring you to the list of tasks that you have in it.<br>
```
Something does not look right here... and which colours are for which priority?

dylansyj committed:
Edit SampleDataUtil class to provide correct examples to user
Edit 2 things in UI only

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-### 2.2. Adding a person: `add`
+### 2.2. Adding a Task : `add`
+Adds a task to the task manager. <br>
+Format: `add TASK_NAME d/DEADLINE p/PRIORITY_LEVEL [i/ANY_INFO] [t/TAGS]...`
```
Refrain from using command line switches or flags.  Although it is easier to parse this way, part of the project requirements is to make the command flexible and as natural as possible.  We expect a fair amount of work in the parser, which is what makes it different from addressbook4.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+### 2.2. Adding a Task : `add`
+Adds a task to the task manager. <br>
+Format: `add TASK_NAME d/DEADLINE p/PRIORITY_LEVEL [i/ANY_INFO] [t/TAGS]...`
+   > `DEADLINE` will be in the format of "date month (indicated by first three letters) year"
```
Make date and time formats flexible also.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-Examples:
+Shows a list of all the tasks in the task manager.<br>
+> Listing is done automatically when user switches between the tabs.<br>
+> However, user can specify to list all the tasks he has in, which will be shown in a pop up.<br>
```
what do you mean by "all the task he has in"?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+Shows a list of all the tasks in the task manager.<br>
+> Listing is done automatically when user switches between the tabs.<br>
+> However, user can specify to list all the tasks he has in, which will be shown in a pop up.<br>
+> User can also sort it accordingly to the deadlines, priorities, etc.
```
Explain the sorting feature later, if there is one...

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-Format: `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...`
+Examples:
+* add CS3230 Assignment 1 d/8 Jan 2018 p/1 i/How to do? t/School t/CS3230
+* add Buy Milk d/23 Feb 2017 p/4 t/Chores
```
How about floating tasks and timed tasks (with a from and to)?  You have to explicitly support these three types of tasks.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-Format: `edit INDEX [NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]...`
+Examples:
+* `list priority 1`
+* `list priority 2`
```
These two examples look the same?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-  Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@yahoo.com` respectively.
+### 2.4. Editing an existing task: `edit`
+Edits an existing task in the task manager.<br>
+Format: `edit i/INDEX [n/TASK_NAME] [d/DEADLINE] [p/PRIORITY_LEVEL] [i/ANY_INFO] [t/TAGS]...`
```
Where does one obtain the `INDEX` associated with a task (likewise for `delete` and `complete`?  From `list` command?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+* `edit i/1 n/Assignment 2 d/25 Feb 2018 p/2`
+* `edit i/4 n/Exercise`
+
+### 2.5. Shifting tabs: `switch`
```
Explaining `switch` here seems rather abrupt.  May I suggest to moving it to the front, even before `add`?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-  `delete 1`<br>
-  Deletes the 1st person in the results of the `find` command.
+Format: `progress NUMBER_OF_DAYS`
+> NUMBER_OF_DAYS is a interger which indicates the your progress for these past days (excluding today).<br>
```
Check for typo and grammar?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+### 2.8. Track overall progress: `progress`
 
-Examples:
+Shows all the tasks completed over time and also specifying the tasks overdued.<br>
```
overdue.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
 
-### 2.7. Select a person : `select`
+Examples:
+* `progress 7`
```
I don't really understand the progress "feature"... maybe a nice-to-have?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-Examples:
+Format: `search KEYWORD`
+> A valid input needs to be keyed in. No special characters such as ASCII whitespace is allowed.<br>
+> A popout will appear to show you the list of tasks that contains the stated `KEYWORD`
```
You seem to like to use "popouts".  How do you close them?  Consider how Jim would like to deal with them.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+Searches for tasks based on keywords by user's input.<br>
 
-Examples:
+Format: `search KEYWORD`
```
Searching only on one keyword?  How about case-insensitivity?  How about partial search?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-Format: `clear`
+Undos previous action that user made.<br>
+Format : `undo`
+> Any previous valid input will be reversed.
```
Define "valid input"?

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
-Address book data are saved in the hard disk automatically after any command that changes the data.<br>
-There is no need to save manually.
+Upon creation of tasks, the tasks will be automatically saved in the folder where the program is held in.
+> Do not erase the saved data as it will result in a complete loss of data that cannot be recovered by the application itself.
```
If the tasks are saved after every mutable command, then it does not really matter that the data erased, as the next command will save a new file.  Or are you mentioning about erasing the data before executing the program?   

You also missed out on the feature to save in a different file/directory.

hwkchia commented on docs/UserGuide.md:
```diff
@@ -1,186 +1,239 @@
+Upon creation of tasks, the tasks will be automatically saved in the folder where the program is held in.
+> Do not erase the saved data as it will result in a complete loss of data that cannot be recovered by the application itself.
 
 ## 3. FAQ
```
Seems like your FAQ still assumes GUI event-based input like clicking?

hwkchia reviewed (commented):
This User Guide seems to have been hastily hacked out.  Typically you have till the end of the week to polish up your deliverable for review, so you have time to make changes.

Moreover, bear in mind that there is a lot of focus on the command line parser in this project, so try to make the command as flexible, and as natural, as you can.

You can refer to more guidance here. 
https://drive.google.com/open?id=1kAth-pTOzwCJFVjS6KU8O50bCvijDqpb6GsOyJTD0Ds

hwkchia commented:
User Guide has been reviewed.  DevGuide will be reviewed soon, so do not close the PR yet.

dylansyj committed:
V0.1 Milestone (tentative)

limmlingg committed:
Merge branch 'Logic' of https://github.com/CS2103JAN2017-W09-B4/main into UI

limmlingg committed:
Refactor UI parts

limmlingg committed:
Fix test cases and more refactoring

limmlingg committed:
Merge pull request #7 from CS2103JAN2017-W09-B4/UI

V0.1

dylansyj committed:
Edit UserGuide - Changed line 26
Remove UserGuide(Original)

limmlingg committed:
Update badges in README

dylansyj committed:
Refactor clear feature

dylansyj committed:
Fix travis & testcase

dylansyj committed:
Able to add floating task (i.e no requirements of deadline etc)

dylansyj committed:
Resolve failed cases.
Unresolved : Errors in test cases - NullPointerException

dylansyj committed:
Code went crazy, fixed it already

dylansyj committed:
Checkstyle fix

dylansyj committed:
Checkstyle Fix

dylansyj committed:
casd

dylansyj committed:
Fix junit - By removing retarded stuffs from taskcard class

dylansyj committed:
Indentation fix

dylansyj committed:
Edit FindCommandTest

dylansyj committed:
Add in extra test case to improve coverage

dylansyj committed:
Merge pull request #14 from CS2103JAN2017-W09-B4/ModifyAddCommand

Modify add command - Enabled the addition of tasks without restriction to parameters

dylansyj committed:
Undo/Redo Feature - for undo-ing add feature only.
Yet to be done :
for edit, delete, clear, etc: basically the rest of the commands

dylansyj committed:
Merge pull request #24 from CS2103JAN2017-W09-B4/Undo_Feature

Undo/Redo Feature - for undo-ing add feature only. The rest to be continued

dylansyj committed:
Update Undo/Redo Feature for Add/Edit commands.
What's left : Delete and Clear

qixiang-xu committed:
Updated Regex For Better Matching

limmlingg committed:
Update icon

limmlingg committed:
Remove Browser panel

eevyern committed:
Updated UserGuide to incorporate some of the comments by Henry

eevyern committed:
Merge pull request #26 from CS2103JAN2017-W09-B4/Documentation

Updated UserGuide to incorporate some of the comments by Henry

limmlingg committed:
Delete BrowserPanel.java

limmlingg committed:
Change font to Comfortaa

dylansyj committed:
Merge pull request #27 from CS2103JAN2017-W09-B4/UI

Enhance GUI

qixiang-xu committed:
Updated Regex for Better Matching (Fix)

dylansyj committed:
Merge pull request #28 from CS2103JAN2017-W09-B4/Flexibility

Updated Regex For Better Matching

dylansyj committed:
Undo function works for Add/Delete/Edit.

dylansyj committed:
Update UserGuide.md

qixiang-xu committed:
Update UserGuide.md

eevyern committed:
Edited UserGuide.md - 2.1 to 2.5.

eevyern committed:
Further edits of UserGuide.md - 2.1 to 2.5.

qixiang-xu committed:
Merge pull request #31 from CS2103JAN2017-W09-B4/QixiangUserGuide

Update UserGuide.md

eevyern committed:
Merge branch 'master' into IanUserGuide

eevyern committed:
Merge pull request #32 from CS2103JAN2017-W09-B4/IanUserGuide

Edits to UserGuide.md - 2.1 to 2.5

limmlingg committed:
Update the second half of UserGuide.md

From section 2.6 to the end

eevyern committed:
Fixing indentation issues in UserGuide.md - 1.6.

eevyern committed:
Merge branch 'IanUserGuide' of https://github.com/CS2103JAN2017-W09-B4/main into IanUserGuide

limmlingg committed:
Remove typo in UserGuide.md

eevyern committed:
Merge pull request #33 from CS2103JAN2017-W09-B4/IanUserGuide

	Fixing indentation issues in UserGuide.md - 1.6.

limmlingg committed:
Merge branch 'master' into MiaoLingUserGuide

eevyern committed:
Grammar edits to UserGuide.md - Intro.

eevyern committed:
Merge pull request #35 from CS2103JAN2017-W09-B4/IanUserGuide

Grammar edits to UserGuide.md - Intro.

dylansyj committed:
Update Developer Guide

qixiang-xu committed:
Update DeveloperGuide.md

limmlingg committed:
Merge pull request #34 from CS2103JAN2017-W09-B4/MiaoLingUserGuide

Update second half of UserGuide.md

dylansyj committed:
Merge pull request #36 from CS2103JAN2017-W09-B4/dylansyj-patch-1

Update Developer Guide

dylansyj committed:
Merge pull request #37 from CS2103JAN2017-W09-B4/QixiangUserGuide

Update DeveloperGuide.md

dylansyj committed:
Undo Feature completed. Works for Add/Edit/Delete/Clear.
Several more Junit test cases to be added soon

dylansyj committed:
Merge branch 'master' into Undo_Edit

eevyern committed:
UserGuide cleaned for consistency.

dylansyj committed:
Update DeveloperGuide.md

dylansyj committed:
Merge pull request #39 from CS2103JAN2017-W09-B4/documentationGrammarCheck

UserGuide cleaned for consistency.

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Fix Junit

dylansyj committed:
Merge pull request #25 from CS2103JAN2017-W09-B4/Undo_Edit

Update undo feature : Enabled for add & edit

dylansyj committed:
Added Collate. Marked out the parts for author A0139161J (Dylan).

dylansyj committed:
Merge pull request #40 from CS2103JAN2017-W09-B4/Code_Quality___Dylan

Added Collate. Marked out the parts for author A0139161J (Dylan).

limmlingg committed:
Add NotificationUtil.java

dylansyj committed:
I AM GOING CRAYYYYY WITH THE RED DAYSSS

qixiang-xu committed:
Implemented "Upcoming" Feature

As a user, I want to find upcoming tasks.

qixiang-xu committed:
DeadLineTest Error Fix

qixiang-xu committed:
Add Test Cases

qixiang-xu committed:
Merge pull request #42 from CS2103JAN2017-W09-B4/FlexiCommand

Implemented "Upcoming" Feature

limmlingg committed:
Add system tray notifications that appear every 2 minutes. Notification will tell the user that a task is due today.

eevyern committed:
Implemented FindTagCommand and all related changes to enable it.

eevyern committed:
Merge pull request #44 from CS2103JAN2017-W09-B4/FindTagCommand

Implemented FindTagCommand and all related changes to enable it.

dylansyj committed:
Add in Complete feature
Remodeled UI a little

dylansyj committed:
Included full complete/uncomplete feature.
Remodeled UI to incorporate complete feature

dylansyj committed:
Merge pull request #43 from CS2103JAN2017-W09-B4/12-reminders-task-due-soon

Implement reminders for tasks

dylansyj committed:
Merge branch 'master' into UI_Remodeling

dylansyj committed:
Quickfix.

dylansyj committed:
Storage fix

dylansyj committed:
Need to fix storage in future @ dylan

dylansyj committed:
Merge pull request #45 from CS2103JAN2017-W09-B4/UI_Remodeling

Partial UI Modeling & Complete feature

dylansyj committed:
Remodel UI

dylansyj committed:
Author indication

dylansyj committed:
Temporary fix to GUITest

limmlingg committed:
Fix failing GUI tests

limmlingg committed:
Fix header IDs in DeveloperGuide.md

limmlingg committed:
Update non functional requirements in DeveloperGuide.md

limmlingg committed:
Add Product Survey for Google Tasks to DeveloperGuide.md

limmlingg committed:
Remove example user stories in DeveloperGuide.md

limmlingg committed:
Update jdk version in prerequisites of DeveloperGuide.md

limmlingg committed:
Update java version in DeveloperGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update DeveloperGuide.md

dylansyj committed:
Refactoring folders

dylansyj committed:
Update DeveloperGuide.md

dylansyj committed:
Merge pull request #49 from CS2103JAN2017-W09-B4/Dev-Guide-Updates

Update DeveloperGuide.md

dylansyj committed:
Minor changes to UI

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Merge pull request #50 from CS2103JAN2017-W09-B4/dylansyj-patch-1

Update UserGuide.md

dylansyj committed:
Merge pull request #51 from CS2103JAN2017-W09-B4/dylansyj-patch-2

Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Merge pull request #52 from CS2103JAN2017-W09-B4/dylansyj-patch-3

Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

dylansyj committed:
Update UserGuide.md

eevyern committed:
Added Wunderlist to the product survey on DeveloperGuide.md

eevyern committed:
Merge pull request #53 from CS2103JAN2017-W09-B4/ProductSurveryWunderlist

Added Wunderlist to the product survey on DeveloperGuide.md

limmlingg committed:
Merge pull request #48 from CS2103JAN2017-W09-B4/developer-guide-updates

Update DeveloperGuide.md
